Sources/Apodini/Semantic Model Builder/Model/Endpoint.swift

```diff
@@ -123,102 +130,108 @@ struct Endpoint<H: Handler>: AnyEndpoint {
-            endpoint.treeNode = self
+
+            endpoint.onInserted(at: self)
             endpoints[endpoint.operation] = endpoint
         } else {
-            var pathComponents = paths
-            if let first = pathComponents.removeFirst() as? _PathComponent {
-                var child = nodeChildren[first.description]
-                if child == nil {
-                    // as we create a new child node we need to check if there are colliding path parameters
-                    if let result = PathComponentAnalyzer.analyzePathComponentForParameter(first) {
-                        if result.parameterMode != .path {
-                            fatalError("Parameter can only be used as path component when setting .http() parameter option to .path!")
-                        }
-                        
-                        if childContainsPathParameter { // there are already some children with a path parameter on this level
-                            fatalError("When inserting endpoint \(endpoint.description) we encountered a path parameter collision on level n-\(pathComponents.count): "
-                                        + "You can't have multiple path parameters on the same level!")
-                        } else {
-                            childContainsPathParameter = true
-                        }
+            let next = context.nextPath()
+            var child = nodeChildren[next]
+
+            if child == nil {
+                // as we create a new child node we need to check if there are colliding path parameters
+                switch next {
+                case .parameter:
+                    if childContainsPathParameter { // there are already some children with a path parameter on this level
+                        fatalError("When inserting endpoint \(endpoint.description) we encountered a path parameter collision on level n-\(context.pathCount): "
+                            + "You can't have multiple path parameters on the same level!")
+                    } else {
+                        childContainsPathParameter = true
                     }
-                    
-                    child = EndpointsTreeNode(path: first, parent: self)
-                    nodeChildren[first.description] = child
+                default:
+                    break
                 }
-                
-                // swiftlint:disable:next force_unwrapping
-                child!.addEndpoint(&endpoint, at: pathComponents)
-            } else {
-                fatalError("Encountered PathComponent which isn't a _PathComponent!")
+
+                child = EndpointsTreeNode(path: next, parent: self)
+                nodeChildren[next] = child
             }
+
+            // swiftlint:disable:next force_unwrapping
+            child!.addEndpoint(&endpoint, context: &context)
```

Comment:
You could also use `if let` to remove force unwrapping. Then you also don't need to check `child` for `nil` in line 190.

Reply:
I did the requested refactoring as you suggested and rediscovered why I wrote it that way about a month ago:
If you use if let you end up with the need of calling `addEndpoint` twice or the need to declare a additional variable (basically a if let where the body just sets the variable 🤔 I thought both were ugly and went for force unwrap. But I try to find a neat way of handling that maybe.

Reply:
I tried some stuff, but I think the approach using force_unwrap is honestly the nicest way of writing that function.
Using `if let`, I either have to define a kind of "temp variable", and the `if let` body pretty much consist of assigning that variable, or I would need to have the `addEndpoint` call twice, which isn't a improvement either.
Thus I think the force_unwrapping approach depicts the control flow best. But feel free to leave any change suggestions if you find a better way 👍

Reply:
What about optional chaining?

```suggestion
            child?.addEndpoint(&endpoint, context: &context)
```

Reply:
@theMomax I would argue against that. It is semantically incorrect (being a "execute this if it is the case that child is available" vs "the child is to be inserted here") and silently ignores any potential regressions made in the code above.

If the majority is for removing that force_unwrap and you guys can agree on a replacement for this I'm all for it 👍. Just stating my opinion here.

Reply:
I could maybe see the following working, by (ab)using the `??` operator with a closure

```swift
let next = context.nextPath()

let child = nodeChildren[next] ?? {
    // as we create a new child node we need to check if there are colliding path parameters
    switch next {
    case .parameter:
        if childContainsPathParameter { // there are already some children with a path parameter on this level
            fatalError("When inserting endpoint \(endpoint.description) we encountered a path parameter collision on level n-\(context.pathCount): "
                + "You can't have multiple path parameters on the same level!")
        } else {
            childContainsPathParameter = true
        }
    default:
        break
    }

    let child = EndpointsTreeNode(path: next, parent: self)
    nodeChildren[next] = child
    return child
}()

child.addEndpoint(&endpoint, context: &context)
```

Not sure if this really an improvement to code readability though.

Reply:
Here is my take on this:
```
let child: EndpointsTreeNode

if let nextChild = nodeChildren[next] {
   child = nextChild
} else {
   // as we create a new child node we need to check if there are colliding path parameters
   switch next {
   case .parameter:
      if childContainsPathParameter { // there are already some children with a path parameter on this level
         fatalError("When inserting endpoint \(endpoint.description) we encountered a path parameter collision on level n-\(context.pathCount): "
                            + "You can't have multiple path parameters on the same level!")
      } else {
         childContainsPathParameter = true
      }
      default:
         break
      }

   child = EndpointsTreeNode(path: next, parent: self)
   nodeChildren[next] = child
}

child.addEndpoint(&endpoint, context: &context)

Reply:
Okay, who created a poll 😅😂

Reply:
I'm also fine with the original version